`ks generate` panics "runtime error: index out of range" instead of printing usage

**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
/kind bug

**What happened**:
```
$ ks init guestbook --namespace ks-dev --env dev --api-spec version:v1.9.4
INFO Using context "minikube" from kubeconfig file "/Users/jsuen/.kube/config"
INFO Creating environment "dev" with namespace "ks-dev", pointing to cluster at address "https://192.168.64.31:8443"
INFO Generating ksonnet-lib data at path '/Users/jsuen/ws/myapps/tutorial/guestbook/lib/v1.9.4'

$ cd guestbook/

$ ks generate
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/ksonnet/ksonnet/actions.(*PrototypeUse).Run(0xc4203076d0, 0xc4203076d0, 0x0)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/actions/prototype_use.go:83 +0x633
github.com/ksonnet/ksonnet/actions.RunPrototypeUse(0xc420386270, 0xc420013e60, 0x1c)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/actions/prototype_use.go:37 +0x5c
github.com/ksonnet/ksonnet/cmd.runAction(0x1c, 0xc420386270, 0x1f36aa1, 0x9)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/cmd/actions.go:111 +0x66
github.com/ksonnet/ksonnet/cmd.glob..func33(0x2d02ca0, 0x2d37948, 0x0, 0x0, 0x0, 0x0)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/cmd/prototype_use.go:47 +0x1c9
github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra.(*Command).execute(0x2d02ca0, 0x2d37948, 0x0, 0x0, 0x2d02ca0, 0x2d37948)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra/command.go:753 +0x468
github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x2d03880, 0x13, 0x0, 0x0)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra/command.go:843 +0x30a
github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra.(*Command).Execute(0x2d03880, 0xc4200b8058, 0x0)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra/command.go:791 +0x2b
main.main()
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/main.go:35 +0x85

```

**What you expected to happen**:
`ks generate` is an incomplete command and should print usage information


**Environment**:
- ksonnet version (use `ks version`):

```
$ ks version
ksonnet version: 0.10.0-alpha.1
jsonnet version: v0.10.0
client-go version: 1.8
```

--- a/actions/prototype_use.go
+++ b/actions/prototype_use.go
@@ -81,6 +81,10 @@ func (pl *PrototypeUse) Run() error {
 		return err
 	}
 
+	if len(pl.args) == 0 {
+		return errors.New("prototype name was not supplied as an argument")
+	}
+
 	query := pl.args[0]
 
 	p, err := findUniquePrototype(query, prototypes)

--- a/e2e/generate_test.go
+++ b/e2e/generate_test.go
@@ -30,6 +30,18 @@ var _ = Describe("ks generate", func() {
 		a = e.initApp(nil)
 	})
 
+	FDescribe("with no arguments", func() {
+		var o *output
+
+		JustBeforeEach(func() {
+			o = a.runKs("generate")
+		})
+
+		It("returns an error", func() {
+			assertExitStatus(o, 1)
+		})
+	})
+
 	Context("deployed-service", func() {
 		It("generates a component using the prototype", func() {
 			o := a.runKs(

